axon-framework/monitoring/tracing.md

```diff
@@ -0,0 +1,535 @@
+| `${CommandBusClass}.dispatch(${commandName})`   | The bus is dispatching the command locally (If using Axon Server: this is invoked by the `AxonServerCommandBus`.                                                               |
+| `${CommandBusClass}.handle(${commandName})`     | The bus is invoking the handler locally.                                                                                                                                       |
+| `${RepositoryClass}.load ${identifier}`         | The aggregate is being loaded by the repository. During this time Axon Framework will obtain a lock, fetch snapshots and events from the event store to hydrate the aggregate. |
+| `LockingRepository.obtainLock`                  | The repository is obtaining a lock for the aggregate. This taking some time indicates that the command was queued due to another command being handled for the same aggregate. |
+
+During handling of commands, other functionality might be invoked such as scheduling deadlines or publishing events. 
+Please refer to the specific sections of this functionality for more information.
+
+### Events
+
+Publishing events during command execution happens in two steps:
+First, adding the events to a queue in the `UnitOfWork` to be committed when the transaction succeeds,
+and second, committing them and writing them to the event store.
+
+| Trace name                                | Description                                                                      |
+|-------------------------------------------|----------------------------------------------------------------------------------|
+| `${EventBusClass}.publish(${EventClass})` | For each event, a short span is created to indicate that an event was published. |
+| `${EventBusClass}.commit`                 | Indicates events being committed to the event store.                             |
+
+The events have their own specific publishing span to link other spans to it that are related.
+Any streaming event processor or saga handling the event in the future will be linked
+to publishing spans, allowing easy click-through.
+
+### Event processors
+
+Event processor invocations are traced as well. They will add the following traces:
+
+| Trace name                                                  | Description                                                                 |
+|-------------------------------------------------------------|-----------------------------------------------------------------------------|
+| `${ProcessorType}[${processorName}](${EventClass})`         | Root trace of handling the event, includes all interceptor invocations.     |
+| `${ProcessorType}[${processorName}].process(${EventClass})` | Inner span of handling the event, after all interceptors have been invoked. |
+
+In case of a `SubscribingEventProcessor`, the invocation happens during command execution.
```

Comment:
This is also not restricted to command execution, due to the AMQP and Kafka Extension. Please provide those options here too.